### PR TITLE
Added GetUserById and GetAllUsers APIs

### PR DESCRIPTION
**Pull request by Krishna92Soni**

### CODE REVIEW

File: APTITUDE/APTITUDETEST.Data/Data/Users/UserRepository.cs [hunk 1]
```diff
@@ -1,13 +1,74 @@
...
+                    StatusCode = ResponseStatusCode.InternalServerError
+                });
+            }
+        }
+
+        public async Task<JsonResult> GetUserById(int id)
+        {
+            try
+            {
+                string sqlQuery = $"Select * from SearchUsers({id},NULL,NULL,NULL,NULL,NULL,NULL,NULL)";
```
Comment (mittalvyas1) [inline APTITUDE/APTITUDETEST.Data/Data/Users/UserRepository.cs:51]: Why Inline Query??

File: APTITUDE/APTITUDETEST.Data/Data/Users/UserRepository.cs [hunk 2]
```diff
@@ -1,13 +1,74 @@
...
+        }
+        public async Task<JsonResult> GetAllUsers(string? searchQuery, int? currentPageIndex, int? pageSize)
+        {
+            try
+            {
+                string sqlQuery = $"Select * from SearchUsers(NULL,'{searchQuery}',NULL,NULL,NULL,NULL,{currentPageIndex},{pageSize})";
+                var userVM = await Task.FromResult(_appDbContext.UsersDetails.FromSqlRaw(sqlQuery).ToList());
+                return new JsonResult(new ApiResponse<List<UserViewModel>>
+                {
+                    Data = userVM,
```
Comment (mittalvyas1) [inline APTITUDE/APTITUDETEST.Data/Data/Users/UserRepository.cs:29]: why we required this much fields in getall view model?

File: APTITUDE/APTITUDETEST.Data/Data/Users/UserRepository.cs [hunk 3]
```diff
@@ -1,13 +1,74 @@
...
+
+        public UserRepository(AppDbContext appDbContext) : base(appDbContext)
         {
+            _appDbContext = appDbContext;
+        }
+        public async Task<JsonResult> GetAllUsers(string? searchQuery, int? currentPageIndex, int? pageSize)
+        {
+            try
+            {
+                string sqlQuery = $"Select * from SearchUsers(NULL,'{searchQuery}',NULL,NULL,NULL,NULL,{currentPageIndex},{pageSize})";
```
Comment (mittalvyas1) [inline APTITUDE/APTITUDETEST.Data/Data/Users/UserRepository.cs:25]: Inline query?????

File: APTITUDE/APTITUDETEST.Data/Data/Users/UserRepository.cs (hunk 1 above)
Comment (Krishna92Soni) [reply APTITUDE/APTITUDETEST.Data/Data/Users/UserRepository.cs:51]: I have called function from postgresql. Its the syntax. Same as storedprocedure.

File: APTITUDE/APTITUDETEST.Data/Data/Users/UserRepository.cs (hunk 2 above)
Comment (Krishna92Soni) [reply APTITUDE/APTITUDETEST.Data/Data/Users/UserRepository.cs:29]: Howmany columns are required ?

File: APTITUDE/APTITUDETEST.Data/Data/Users/UserRepository.cs (hunk 3 above)
Comment (Krishna92Soni) [reply APTITUDE/APTITUDETEST.Data/Data/Users/UserRepository.cs:25]: It is the syntax for calling function from postgres.

File: APTITUDE/APTITUDETEST.Core/Interfaces/IUsersRepository.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using APTITUDETEST.Core.Entities.Users;
+using Microsoft.AspNetCore.Mvc;
+
+namespace AptitudeTest.Core.Interfaces
+{
+
+    public interface IUsersRepository : IRepositoryBase<User>
```
Comment (ShubhamPadiyaTatvasoft) [inline APTITUDE/APTITUDETEST.Core/Interfaces/IUsersRepository.cs:7]: is base repository is not required then remove.

File: APTITUDE/APTITUDETEST.Data/Data/UserRepository.cs [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+using APTITUDETEST.Core.Entities.Users;
+using Dapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Npgsql;
+
+namespace AptitudeTest.Data.Data
+{
+    public class UserRepository : RepositoryBase<User>, IUsersRepository
```
Comment (ShubhamPadiyaTatvasoft) [inline APTITUDE/APTITUDETEST.Data/Data/UserRepository.cs:14]: remove base class

File: APTITUDE/APTITUDETEST.Data/Data/UserRepository.cs [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+        #endregion
+
+        #region methods
+
+        #region GetAllUsers
+        public async Task<JsonResult> GetAllUsers(string? searchQuery, int? currentPageIndex, int? pageSize)
+        {
+            try
+            {
+                List<User> AllUserData = await Task.FromResult(_appDbContext.Users.Include(user => user.MasterGroups).Include(user => user.MasterTechnologies).ToList());
```
Comment (ShubhamPadiyaTatvasoft) [inline APTITUDE/APTITUDETEST.Data/Data/UserRepository.cs:40]: remove linq query

**Review by ShubhamPadiyaTatvasoft — APPROVED** [review]

File: APTITUDE/APTITUDETEST.Data/Data/UserRepository.cs [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+                                userDetails.AcademicsDetails.Add(userAcademics);
+                            }
+                            AcadamicIds.Add(entity.AcademicId);
+                        }
+
+                        if (entity.familyid != null)
+                        {
+                            if (!FamilyIds.Contains(entity.familyid))
+                            {
+                                UserFamilyVM family = new UserFamilyVM();
```
Comment (mittalvyas1) [inline APTITUDE/APTITUDETEST.Data/Data/UserRepository.cs:168]: Split into small functions

Comment (meetkumar-tatva) [reply APTITUDE/APTITUDETEST.Data/Data/UserRepository.cs]: Done

**Review by mittalvyas1 — APPROVED** [review]

**Review by mittalvyas1 — APPROVED** [review]

**Pull request merged**